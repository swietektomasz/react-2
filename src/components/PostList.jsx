import React, { useState, useEffect } from 'react';
import { windowFetch } from '../api';
import GenericPost from './GenericPost';
import ColorfulPost from './ColorfulPost';
import BorderedPost from './BorderedPost';

function PostList() {
  const [data, setData] = useState({ posts: [] });

  useEffect(() => {
    windowFetch().then((newPosts) => {
      setData({ posts: newPosts });
    });
  }, []);

  return (
    <div>
      <h2>A generic list of posts</h2>
      {data.posts.map((post) => <GenericPost key={post.id} post={post} />)}
      <h2>A post colored blue</h2>
      <ColorfulPost post={data.posts[0]} />
      <h2>A post with a border</h2>
      <BorderedPost post={data.posts[0]} />
    </div>
  );
}

export default PostList;
