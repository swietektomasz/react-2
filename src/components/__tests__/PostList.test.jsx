import React from 'react';
import { render } from '@testing-library/react';
import 'whatwg-fetch';

import PostList from '../PostList';

const mockedPostsResponse = [
  { id: 1, title: 'test post 1', author: 'test' },
  { id: 2, title: 'test post 2', author: 'test' },
];

test('renders the PostList component correctly', () => {
  jest.spyOn(window, 'fetch').mockImplementationOnce(() => Promise.resolve({
    json: () => Promise.resolve(mockedPostsResponse),
  }));

  const { container } = render(<PostList />);
  expect(container.firstChild).toMatchSnapshot();
});
