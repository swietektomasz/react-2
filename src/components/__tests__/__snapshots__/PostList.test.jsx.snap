// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PostList component correctly 1`] = `
<div>
  <h3>
    A list of all the posts
  </h3>
  <ol />
</div>
`;
